File: Sort/meeting2/Solution.py

# Support code
class Event:
    def __init__(self, time, is_start):
        self.time = time
        self.is_start = is_start

    def __lt__(self, other):
        if self.time == other.time:
            return not self.is_start
        return self.time < other.time


# Problem code
def minMeetingRooms(intervals):

    events = []
    for interval in intervals:
        events.append(Event(interval[0], True))
        events.append(Event(interval[1], False))
    count = 0
    maximum = 0
    for event in sorted(events):
        if event.is_start:
            count += 1
        else:
            count -= 1
        maximum = max(maximum, count)
    return maximum

File: Sort/meeting2/Solution.swift

// Support code
struct Event {
    var time: Int
    var isStart: Bool
}

// Problem code
func minMeetingRooms(_ intervals: [[Int]]) -> Int {
        
	var events = Array<Event>()
	for interval in intervals {
		events.append(Event(time: interval[0], isStart: true))
		events.append(Event(time: interval[1], isStart: false))
	}
	events.sort() {
		if $0.time == $1.time {
			return !$0.isStart
		}
		return $0.time < $1.time
	}
	
	var count = 0
	var maximum = 0
	
	for event in events {
		if event.isStart {
			count += 1
		} else {
			count -= 1
		}
		maximum = max(maximum, count)
	}
	return maximum
}


